src/VisualStudio/Core/Def/Implementation/Remote/VisualStudioRemoteHostClientProvider.cs

```diff
@@ -38,23 +45,42 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
                     return new DefaultRemoteHostClientProvider();
                 }
 
-                return new VisualStudioRemoteHostClientProvider(workspaceServices);
+                return new VisualStudioRemoteHostClientProvider(workspaceServices, _vsServiceProvider);
             }
         }
 
         private readonly HostWorkspaceServices _services;
-        private readonly AsyncLazy<RemoteHostClient> _lazyClient;
+        private readonly AsyncLazy<RemoteHostClient?> _lazyClient;
+        private readonly IAsyncServiceProvider _vsServiceProvider;
 
-        private VisualStudioRemoteHostClientProvider(HostWorkspaceServices services)
+        private VisualStudioRemoteHostClientProvider(HostWorkspaceServices services, IAsyncServiceProvider vsServiceProvider)
         {
             _services = services;
-            _lazyClient = new AsyncLazy<RemoteHostClient>(CreateHostClientAsync, cacheResult: true);
+            _vsServiceProvider = vsServiceProvider;
+            _lazyClient = new AsyncLazy<RemoteHostClient?>(CreateHostClientAsync, cacheResult: true);
         }
 
-        private Task<RemoteHostClient> CreateHostClientAsync(CancellationToken cancellationToken)
-            => ServiceHubRemoteHostClient.CreateAsync(_services, cancellationToken);
+        private async Task<RemoteHostClient?> CreateHostClientAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                var brokeredServiceContainer = await _vsServiceProvider.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>().ConfigureAwait(false);
+                var serviceBroker = brokeredServiceContainer.GetFullAccessServiceBroker();
+
+                var client = await ServiceHubRemoteHostClient.CreateAsync(_services, serviceBroker, cancellationToken).ConfigureAwait(false);
+
+                // proffer in-proc brokered services:
+                _ = brokeredServiceContainer.Proffer(SolutionAssetProvider.ServiceDescriptor, (_, _, _, _) => new ValueTask<object?>(new SolutionAssetProvider(_services)));
```

Comment:
does this basically mean "teh oop side will be able to reach this VS-inproc service"?

Reply:
yup